core/extraction/towns_fund.py

```diff
@@ -65,11 +66,47 @@ def ingest_towns_fund_data(df_ingest: pd.DataFrame) -> Tuple[Dict[str, pd.DataFr
         project_lookup,
         programme_id,
     )
+    towns_fund_extracted["Submission_Ref"] = pd.DataFrame(
+        data=[
+            {
+                "Submission ID": 1,
+                "Submission Date": datetime.datetime.now(),
+                "Reporting Period Start": datetime.datetime.now(),
+                "Reporting Period End": datetime.datetime.now(),
+                "Reporting Round": 1,
+            }
+        ]
+    )
+
+    add_submission_ids(towns_fund_extracted)
+
     reporting_period = df_ingest["1 - Start Here"].iloc[4, 1]
 
     return towns_fund_extracted, reporting_period
 
 
+def add_submission_ids(dfs):
+    tables_need_sub_ids = [
```

Comment:
We might not be able to do it like this at extraction time, as we'll only create the submission ID when we load into the DB (as it needs to check the DB for the previous one). We need to think about transaction control to prevent race conditions

Reply:
Unless you propose using the dummy id for linking the entities, then updating the submission_id afterwards?

Reply:
This is just a hacky solution to single ingest - wasn't proposing anything. That being said, that doesn't sound like a bad idea on the face of it?

Reply:
this is now properly implemented by a prior PR